windows-install-anywhere.md

```diff
@@ -0,0 +1,108 @@
+- Feature Name: Windows Install Anywhere
+- Start Date: 2021-01-19
+- SEP Status: Draft
+- SEP PR: (leave this empty)
+- Salt Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Allow the user to select the install location for salt in the Windows Installer.
+Default to Program Files
+
+# Motivation
+[motivation]: #motivation
+
+Windows is not Linux
+
+The Windows directory structure is such that installed software should reside in
+the ``Program Files`` directory on the ``C:`` drive. Software should not be
+installed on the root of the system directory. This seems to be a paradigm
+brought over from Linux.
+
+- Non-changing binaries should be in ``Program Files``
+- Volatile data should be in ``ProgramData``
+- Configuration should be in the registry, though we would probably make an
+  exception for salt due to its reliance on a config file
+- Some config could be placed in the registry to tell salt where the config file
+  resides
+
+We want the salt installer to be consistent with other enterprise installers
+that follow the Windows paradigm for Software installation
+
+# Design
+[design]: #detailed-design
+
+In order for this to work we will have to separate out much of the salt code. We
+need to separate binary and volatile data as well as config data. It is proposed
+that it is broken out according to the following:
+
+Binary Data:
+
+c:\salt\*.files
+-> C:\Program Files\<salt>\*.files
+
+c:\salt\bin
+-> C:\Program Files\<salt>\bin
+
+Volatile Data:
+
+c:\salt\conf
+-> C:\ProgramData\<salt>\conf
+
+c:\salt\var
+-> C:\ProgramData\<salt>\var
+
+c:\salt\srv
+- C:\ProgramData\<salt>\srv
+
+There is some discussion to be had over the name of the <salt> directory. Here
+are a few proposals:
+
+- Salt
+- SaltProject
+- Salt Project **
+- saltproject.io
+
+We recommend ``Salt Project``
```

Comment:
Are spaces in directory names on windows a common source of problems?

Reply:
Yes, for poorly-written software. Having spaces by default might be good for uncovering any bugs that get be fixed.

Reply:
Yeah our **number 1 goal** with this SEP is to create as much consistency as possible based off typical installations of software for Windows/MS.

Reply:
Note that "Program Files" already has a space in it, so another space in "Salt Project" isn't going to make any difference.

Reply:
Just trying to raise awareness.

@orangedog, `c:/salt` doesn't have spaces, that's the current default.

Of course, if you add one space to the path, 2 or 3 more won't hurt :)

Reply:
Salt *shouldn't* have any issues with being installed in a different location, since we should be using all of the builtin Python pathlib/os.path functions.